File: src/compiler/typescript_to_rust/generators/ic_object/functions/data_certificate.ts
import { Rust } from '../../../../../types';

export function generateIcObjectFunctionDataCertificate(): Rust {
    return /* rust */ `
        fn _azle_ic_data_certificate(
            _this: &boa_engine::JsValue,
            _aargs: &[boa_engine::JsValue],
            _context: &mut boa_engine::Context
        ) -> boa_engine::JsResult<boa_engine::JsValue> {
            Ok(ic_cdk::api::data_certificate().azle_into_js_value(_context))
        }
    `;
}

File: src/compiler/typescript_to_rust/azle_generate/src/body/ic_object/functions/data_certificate.rs
pub fn generate() -> proc_macro2::TokenStream {
    quote::quote! {
        fn data_certificate(
            _this: &boa_engine::JsValue,
            _aargs: &[boa_engine::JsValue],
            context: &mut boa_engine::Context
        ) -> boa_engine::JsResult<boa_engine::JsValue> {
            Ok(ic_cdk::api::data_certificate().try_into_vm_value(context).unwrap())
        }
    }
}
